llvm/utils/TableGen/DXILEmitter.cpp

```diff
@@ -151,7 +139,8 @@ DXILOperationDesc::DXILOperationDesc(const Record *R) {
 DXILParameter::DXILParameter(const Record *R) {
   Name = R->getValueAsString("Name");
   Pos = R->getValueAsInt("Pos");
-  Kind = parameterTypeNameToKind(R->getValueAsString("Type"));
+  Kind = getDXILTypeNameToKind(
+      R->getValue("ParamType")->getValue()->getAsString());
```

Comment:
The name `getDXILTypeNameToKind` is a bit confusing, both because it doesn't "get" a "DXILTypeName" and because "get" usually implies an accessor or something that's otherwise pretty free. Maybe something like "lookupParameterKind" would be better.

Reply:
Changed. Thanks!